### PR TITLE
show default value parameters in Outline

### PR DESCRIPTION
**Pull request by nyrosmith**
Associated Issue: #5265

default_paramst

### CODE REVIEW

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap [hunk 1]
```diff
@@ -841,97 +841,28 @@ hasJsx: false"
 
 exports[`Parser.getSymbols function names 1`] = `
 "functions:
```
Comment (jasonLaster) [inline src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: why are we losing the snapshot data?

File: src/workers/parser/getSymbols.js [hunk 2]
```diff
@@ -44,7 +44,18 @@ export type SymbolDeclarations = {
...
+        return param.name;
+      } else if (
+        // Parameter with default value
+        param.type === "AssignmentPattern" &&
+        param.left.type === "Identifier" &&
+        param.right.type === "Identifier"
+      ) {
+        return `${param.left.name} = ${param.right.name}`;
+      }
+    });
```
Comment (jasonLaster) [inline src/workers/parser/getSymbols.js]: is there now a case that we're not handling?

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap (hunk 1 above)
Comment (nyrosmith) [reply src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: I don't know I just updated the snapshots to reflect the new test case

File: src/workers/parser/getSymbols.js (hunk 2 above)
Comment (nyrosmith) [reply src/workers/parser/getSymbols.js]: Yeah. Eg objects, arrays or functions as default values.
Right now I have no idea how these should be displayed

Comment (nyrosmith) [reply src/workers/parser/getSymbols.js]: I'll try to sketch out how to display the other cases over the weekend

**Comment by darkwing** [comment]
I was talking to @jasonLaster yesterday about this same idea!  Great minds!

I was thinking we could do more of an "expand to view args" functionality; one thing I like about this pane is that it's simple.  Thoughts?

**Comment by jasonLaster** [comment]
+1 on keeping it simple!

In the future i'd love to show imports and exports here too!

File: src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap [hunk 3]
```diff
@@ -881,6 +882,8 @@ variables:
 [(25, 13), (25, 31)]   foo
 [(26, 12), (26, 42)]
 [(26, 13), (26, 36)]   foo
```
Comment (jasonLaster) [inline src/workers/parser/tests/__snapshots__/getSymbols.spec.js.snap]: looks like we're missing `first` variable here

File: src/workers/parser/tests/fixtures/functionNames.js [hunk 4]
```diff
@@ -38,3 +38,5 @@ class Cls {
 (function(){});
 
 export default function (){}
+
+function def(first = defaultValue, second){}
```
Comment (jasonLaster) [inline src/workers/parser/tests/fixtures/functionNames.js]: we didn't get any values because `function(first = defaultValue, second){}` is a syntax error

**Comment by nyrosmith** [comment]
default_params

We are now showing much more cases than before. I checked every default assignment I could think of.
Caveats: Inline objects and arrays are always displayed as empty. I couldn't find an appropriate UI representation if they have values...

**Comment by nyrosmith** [comment]
default_params

We are now showing much more cases than before. I checked every default assignment I could think of.
Caveats: Inline objects and arrays are always displayed as empty. I couldn't find an appropriate UI representation if they have values...

File: src/workers/parser/getSymbols.js [hunk 5]
```diff
@@ -44,7 +44,50 @@ export type SymbolDeclarations = {
...
+          param.left.type === "Identifier" &&
+          param.right.type === "NullLiteral"
+        ) {
+          return `${param.left.name} = null`;
+        }
+      } else {
+        return param.name;
+      }
+
+      // if (param.name) {
```
Comment (nyrosmith) [inline src/workers/parser/getSymbols.js]: I forgot to delete this

**Comment by jasonLaster** [comment]
Great work!

**Pull request merged**